### DISCUSSION TITLE
Detecting Full Load of NiceGUI App in Native Mode

### DISCUSSION THREAD

**Discussion by impubes** [zauberzeug/nicegui#3536] [Q&A] [answered]
### Question

Hello,

I am trying to implement a splash screen for a NiceGUI app (with native mode activated), which will be packaged as an .exe file using PyInstaller. I am using the --splash option of PyInstaller to start the splash screen (PNG file).

To detect when the native app is fully loaded, I tried using the app.on_startup event provided by NiceGUI. However, it does not work as expected, as it closes the splash screen too early.

Is there a way to accurately detect when the native mode (pywebview) is fully loaded?

Thank you.

**Comment by rodja** [comment, marked as answer]
You could [wait for the bowser to connect](https://nicegui.io/documentation/page#wait_for_client_connection) but depending on your content (large images, async delays) it will still be to early...

**Reply by impubes** [reply]
`app.on_connect` works perfectly; thank you for your answer.  :)

```python
# ...............

try:
    import pyi_splash
except ImportError:
    pyi_splash = None

# ..................

async def splash_closure():
    if pyi_splash:
        pyi_splash.close()
app.on_connect(splash_closure)

app.native.window_args['resizable'] = False
ui.run(native=True, reload=False, window_size=(650, 358), fullscreen=False)

```
